android/app/src/main/java/org/fossasia/openevent/viewmodels/FeedFragmentViewModel.java

```diff
@@ -0,0 +1,45 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+
+import org.fossasia.openevent.api.APIClient;
+import org.fossasia.openevent.data.facebook.FacebookPageId;
+import org.fossasia.openevent.data.facebook.Feed;
+import org.fossasia.openevent.utils.ConstantStrings;
+import org.fossasia.openevent.utils.SharedPreferencesUtil;
+
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
+public class FeedFragmentViewModel extends ViewModel {
+
+    private MutableLiveData<FacebookPageId> facebookPageIdMutableLiveData;
+
+    public FeedFragmentViewModel() {
+        facebookPageIdMutableLiveData = new MutableLiveData<>();
+    }
+
+    public LiveData<FacebookPageId> updateFBPageID(String fbAccessToken) {
+        if (SharedPreferencesUtil.getString(ConstantStrings.FACEBOOK_PAGE_ID, null) == null)
+            APIClient.getFacebookGraphAPI().getPageId(SharedPreferencesUtil.getString(ConstantStrings.FACEBOOK_PAGE_NAME, null),
+                    fbAccessToken)
```

Comment:
Here, sending LiveData instead of Observable is no good because you are not disposing the observable properly, It will cause memory leak.

Convert Observable to Flowable and then to LiveData through
`LiveDataReactiveStreams.fromPublisher`

Reply:
Done